### PR TITLE
Add rightClick to ariakit-test

### PR DESCRIPTION
**Pull request by diegohaz**
## Motivation

`@ariakit/test` had helpers for primary clicks and low-level event dispatching, but no user-level helper for secondary mouse clicks. Tests for the context menu example were using `dispatch.contextMenu(...)`, which opens the menu but skips the browser-like pointer and mouse sequence around a right click.

## Solution

Add `rightClick(element, options?)` to `@ariakit/test`. The helper hovers the target, presses the secondary pointer button, dispatches `contextmenu` while the button is still pressed, releases it, and dispatches `auxclick` without firing a synthetic primary `click`.

This also retargets `contextmenu` and `auxclick` through the same `pointer-events: none` path used by other pointer/mouse events, covers the new helper with package tests, documents it in the generated README, and migrates the context-menu example tests from `dispatch.contextMenu(...)` to `rightClick(...)`.

### CODE REVIEW

File: packages/ariakit-test/src/right-click.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      element = element.parentElement;
+    }
+
+    await dispatch.contextMenu(element, pressOptions);
+    await settle();
+
+    const releaseOptions = getReleaseOptions(options);
+
+    await mouseUp(element, releaseOptions);
+    await dispatch(element, createAuxClickEvent(element, releaseOptions));
```
[the remark this answers is not in the document]
Comment (diegohaz) [reply packages/ariakit-test/src/right-click.ts:71]: Reviewed this against the current head. The edge case is technically real — if a `contextmenu` handler hides or detaches `element`, `mouseUp()` early-returns on the now-hidden node (`mouse-up.ts` line 25) while line 71 still dispatches `auxclick` on it — but I'd treat it as low priority rather than Major, and I'd lean against applying the fix in isolation:

- **It mirrors `click` by design.** `click()` resolves the visible-parent target once after `mouseDown` (`click.ts:187-190`), then `settle()` → `mouseUp()` → `dispatch.click()` with no re-resolution. So if the target is hidden during the dwell, `click()` likewise skips `mouseup` but still fires `click` on the stale/hidden node. `rightClick` follows the same shape; adding the re-resolution only here would make the two helpers diverge. If this matters, it's better handled uniformly in both helpers as a separate change.
- **The effect is benign.** A detached/hidden node has no live ancestors to bubble to, so the trailing `auxclick` is a harmless no-op rather than a crash or wrong-target dispatch.
- **It doesn't trigger in the migrated use case.** In `examples/menu-context-menu`, the `Right click here` trigger stays in the DOM when the menu opens, so the release target is never hidden.

Validation: `right-click.test.ts` passes on both happy-dom and jsdom (3/3 each), `examples/menu-context-menu/test.ts` is stable across 3 runs, and the full `ariakit-test` suite is 71/71. So no observable failure today; my suggestion is to keep it as-is for parity with `click`.

**Review by diegohaz — COMMENTED** [review]
## Review summary

**Initial review** of the full PR. Baseline: merge base `358d1dd2` → head `f2de2e1`. Scope: all 7 changed files plus the surrounding `@ariakit/test` helpers they build on (`click.ts`, `dispatch.ts`, `mouse-down.ts`, `mouse-up.ts`, `hover.ts`).

**Overall: clean, no blocking findings.** The new `rightClick` faithfully mirrors the existing `click` helper, the design decisions are justified, and behavior is well covered.

### What I verified
- **Manual `auxclick` creation is justified.** `@testing-library/dom`'s event map has `click`/`contextMenu`/`dblClick` but no `auxClick`, so `dispatch.auxClick` doesn't exist — building the event by hand is the right call. Constructing it from the element's `ownerDocument.defaultView.MouseEvent` keeps `event instanceof MouseEvent` working in `baseDispatch` across windows/iframes, and the native `MouseEvent` constructor reflects `button`/`buttons`/`shiftKey` correctly without needing `initEvent` (confirmed by the passing assertions on both engines).
- **`dispatch.ts` allowlist change is safe.** Adding `auxclick`/`contextmenu` to `pointerEvents` is the correct browser behavior (these route around `pointer-events: none`), and the only `dispatch.contextMenu` caller in the repo is `rightClick` itself, so no existing test depended on the old non-retargeting behavior. Full `ariakit-test` suite: 71/71.
- **Event sequence + options.** `pointerdown`/`mousedown`/`focus`/`contextmenu`/`pointerup`/`mouseup`/`auxclick` with `button: 2`, `buttons: 2` on press and `buttons: 0` on release, no synthetic primary `click` — matches the tests and the documented contract.
- **Example migration.** `examples/menu-context-menu/test.ts` passes and is stable across 3 runs. Dropping `expect(q.menu()).toHaveFocus()` on the second (re-trigger-while-open) right-click is a reasonable fidelity adjustment: the realistic press sequence blurs before `contextmenu`, and re-showing an already-open modal menu doesn't synchronously re-grab focus — `toBeVisible` is the meaningful assertion there.
- **README + changeset.** The README is hand-maintained (no generator), the entry is alphabetically placed, and the `@ariakit/test` patch changeset wording matches repo convention. No `examples` changeset is needed (test-only change).

### Tests run locally
- `vitest run right-click` on happy-dom and `ARIAKIT_TEST_ENV=jsdom` — 3/3 each.
- `pnpm test-react run menu-context-menu` — 3/3, repeated 3× (no flakiness).
- `vitest run packages/ariakit-test` — 71/71.
- All PR CI checks green (Type check, Lint, Test React/React18/Solid/Chrome/Firefox/Safari, Build).

### Non-blocking notes
- One open CodeRabbit thread (re-resolve the release target after `settle()`): I replied with an evidence-based reassessment — the edge case is real but benign and intentionally mirrors `click`; my recommendation is to keep it as-is for parity unless it's addressed uniformly in both helpers.
- `rightClick` duplicates `click`'s skeleton (invariant → `isVisible` → `hover` → `mouseDown` → visible-parent walk → `settle` → `mouseUp` → `sleep`). I considered extracting a shared helper but the per-phase option transforms plus the `contextmenu`/`auxclick` divergence would need enough parameterization that two self-contained functions read better — consistent with how `hover`/`mouseDown`/`mouseUp`/`click` are each standalone. Not worth changing.

### Not fully validated
- Behavior on disabled targets (`contextmenu`/`auxclick` currently fire unconditionally, unlike `click`'s `disabled` guard before the terminal event) — no test covers it and real-browser behavior for right-clicking disabled controls is ambiguous, so I didn't treat it as actionable. `rightClick` is aimed at context-menu triggers, which are rarely disabled.
